docs/_releases/v2.0.0-pre.4/matchers.md

```diff
@@ -112,6 +112,26 @@ Requires the value to be a `Function`.
 Requires the value to be an `Array`.
 
 
+#### `sinon.match.array.deepEquals(iterable)`
+
+Requires an iterable value to be deep equal another one.
```

Comment:
I think we should go for the first option too. I think it would make for clearer tests as well, and we won't come in the position of always being late with implementing all the variations of what `iterable` could constitute => simpler implementation, less branching. As a minor thing, it also means we don't have to change the existing code.

Agree/disagree, @fearphage?